rust/datafusion/src/optimizer/filter_push_down.rs

```diff
@@ -215,13 +215,38 @@ fn issue_filters(
     push_down(&state, &plan)
 }
 
+/// converts "A AND B AND C" => [A, B, C]
+fn split_members(predicate: &Expr) -> Vec<&Expr> {
```

Comment:
Eventually (as part of some other PR) this code might more naturally belong in the utils.rs module -- splitting expressions into an AND list is a common thing to want to do, I think. It would make the code easier to find if people need it for other optimizations. I feel like I have seen this code before in DataFusion but I can't find it now...